roboteam_ai/src/control/shotControllers/ShotController.cpp

```diff
@@ -110,55 +108,53 @@ Vector2 ShotController::getPlaceBehindBall(world::Robot robot, Vector2 shotTarge
     return ball->pos + preferredShotVector.stretchToLength(distanceBehindBall);
 }
 
-// use Numtree GTP to go to a place behind the ball
-ShotData ShotController::goToPlaceBehindBall(world::Robot robot, Vector2 robotTargetPosition,
+/// use Numtree GTP to go to a place behind the ball
+RobotCommand ShotController::goToPlaceBehindBall(world::Robot robot, Vector2 robotTargetPosition,
         std::pair<Vector2, Vector2> line) {
     auto ball = world::world->getBall();
-    control::PosVelAngle pva = robot.getNumtreePosControl()->getPosVelAngle(std::make_shared<world::Robot>(robot),
+    auto shotData = robot.getNumtreePosControl()->getPosVelAngle(std::make_shared<world::Robot>(robot),
             robotTargetPosition);
     //TODO: if (rotating to this angle from current angle will hit ball) then pva.angle=angle towards ball
     if ((robot.pos - robotTargetPosition).length() < 0.3) {
-        pva.angle = (line.second - line.first).toAngle();
+        shotData.angle = (line.second - line.first).toAngle();
     }
-
-    ShotData shotData(pva);
     return shotData;
 }
 
 /// At this point we should be behind the ball. now we can move towards the ball to kick it.
-ShotData ShotController::moveStraightToBall(world::Robot robot, std::pair<Vector2, Vector2> lineToDriveOver) {
-    control::PosVelAngle pva = robot.getBasicPosControl()->getPosVelAngle(std::make_shared<world::Robot>(robot),
+RobotCommand ShotController::moveStraightToBall(world::Robot robot, std::pair<Vector2, Vector2> lineToDriveOver) {
+    auto pva = robot.getBasicPosControl()->getPosVelAngle(std::make_shared<world::Robot>(robot),
             lineToDriveOver.second);
     pva.angle = (lineToDriveOver.second - lineToDriveOver.first).angle();
-    ShotData shotData(pva);
+    RobotCommand shotData(pva);
     return shotData;
 }
 
 /// Now we should have the ball and kick it.
-ShotData ShotController::shoot(world::Robot robot, std::pair<Vector2, Vector2> driveLine, Vector2 shotTarget, bool chip,
+RobotCommand ShotController::shoot(world::Robot robot, std::pair<Vector2, Vector2> driveLine, Vector2 shotTarget, bool chip,
         BallSpeed desiredBallSpeed) {
 
     auto ball = world::world->getBall();
 
     // move towards the ball
-    control::PosVelAngle pva = robot.getBasicPosControl()->getPosVelAngle(std::make_shared<world::Robot>(robot),
+    auto pva = robot.getBasicPosControl()->getPosVelAngle(std::make_shared<world::Robot>(robot),
```

Comment:
You should maybe refactor all the pva named variables something else

Reply:
123 fix some